docs/diagrams/data-flow/data-flow-api-endpoint.puml

```diff
@@ -0,0 +1,153 @@
+
+autonumber
+
+box Consumer #ConsumerColor
+    participant CBE as "Backend Service"
+    participant CCP as "Control Plane" #Business
+    participant CDP as "Data Plane" #Business
+end box
+
+box Provider #ProviderColor
+    participant PDP as "Data Plane" #Business
+    participant PCP as "Control Plane" #Business
+    participant PBE as "Backend Service"
+end box
+
+title
+ **Access to REST API**
+ <sub>Control Plane and Date Plane are part of the EDC.</sub>
+ <sub>Precondition: contract 'access to api' iuncluding constraints (e.g. how often, for which purpose etc.) is signed</sub>
+
+ Case 1a: Each API call matches one TransferProcess (http push)
+ <sub>Provider is pushing data to the Consumer</sub>
+end title
+
+CBE ->> CCP : getAsset(query params, int destination)
+activate CCP
+CCP ->> CDP : provision ext endpoint
+CCP ->> PCP : start data transfer(asset, contract, query params, ext endpoint)
+activate PCP
+PCP ->> PBE : provision data\n(if needed)
+PCP ->> PDP : start data transfer
+activate PDP
+PDP -> PBE : pull data
+activate PBE
+return data
+PDP -> CDP : push data to ext endpoint
+activate CDP
+CDP -> CBE : push data to int destination
+note left of CDP : Dataplane\nacts as proxy.
+PDP ->> PCP : notify: data pushed
+deactivate PDP
+CDP ->> CCP : notify: data received
+deactivate CDP
+CCP ->> CDP : deprovision
+PCP ->> PBE : deprovision\n(if needed)
+deactivate CCP
+deactivate PCP
+
+
+
+newpage Case 1b: Each API call matches one TransferProcess (http pull)\n<sub>Consumer is pulling data from Provider</sub>
+
+autonumber 1
+
+CBE ->> CCP : getAsset(query params, int destination)
+activate CCP
+CCP ->> PCP : start data transfer(asset, contract, query params)
+activate PCP
+PCP ->> PBE : provision data\n(if needed)
+PCP ->> PDP : provision ext endpoint
+PCP ->> CCP : ready to transfer (ext endpoint)
+CCP -> CDP : start data transfer
+
+activate CDP
+CDP -> PDP : pull data from ext endpoint
+activate PDP
+PDP -> PBE : pull data
+note right of PDP : Dataplane\nacts as proxy.
+activate PBE
+return data
+return data
+CDP -> CBE : push data to int destination
+PDP ->> PCP : notify: data pulled
+
+CDP ->> CCP : notify: data received
+deactivate PDP
+
+deactivate CDP
+CCP ->> CDP : deprovision
+PCP ->> PBE : deprovision\n(if needed)
+deactivate CCP
+deactivate PCP
+
+
+
+newpage Case 2: Access to API matches TransferProcess
+autonumber 1
+== Initialization ==
+
+CBE ->> CCP : getAsset(callback)
+activate CCP
+CCP ->> CDP : provision
+CCP ->> PCP : start data transfer(asset, contract)
+activate PCP
+PCP ->> PBE : provision
+PCP ->> PDP : provision
+PCP ->> CCP : ready
+CCP ->> CBE : ready(API access)
+
+== Working ==
+note right of CCP : stays in state\nSTREAMING
+note left of PCP : stays in state\nSTREAMING
+CBE -> CDP : API Call (GET/PUT/POST/DELETE/...)
+note left of CDP : Dataplane\nacts as proxy.\nCould adjust calls
+activate CDP
+CDP -> PDP : API Call
+note right of PDP : acts as proxy\nsee Consumer
+activate PDP
+PDP -> PBE : API Call
```

Comment:
I would keep it like in the other pictures:
```suggestion
PDP -> PBE : pull data
```

Reply:
but it's not a "pull" data. Can also be a POST call, a DELETE call etc.  Think of any REST interface.